ucsdscheduleme/Repo/FormatRepo.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ucsdscheduleme.Models;
+
+namespace ucsdscheduleme.Repo
+{
+    public class FormatRepo
+    {
+        private static readonly int START_HOUR = 7;
+        private static readonly int START_MINUTES = 30;
+        private static readonly int MINUTES_IN_HOUR = 60;
+
+        public static ScheduleViewModel FormatSectionsToCalendarEvent(List<Section> selectedSections)
+        {
+            ScheduleViewModel model = new ScheduleViewModel
+            {
+                Courses = new Dictionary<int, CourseViewModel>()
+            };
+            var coursesSectionPairs = selectedSections.ToDictionary(s => s,s => s.Course);
+
+            foreach (var courseSection in coursesSectionPairs)
+            {
+                var course = courseSection.Value;
+                var selectedSection = courseSection.Key;
+                CourseViewModel thisCourse = new CourseViewModel
+                {
+                    Bases = new Dictionary<char, BaseViewModel>(),
+                    SelectedSection = selectedSection.Id,
+                    // If our sections have no meetings, we're in trouble. Time to abort.
+                    SelectedBase = selectedSection.Meetings?.First().Code[0] ?? 
+                        throw new ArgumentException($"Section with no meeting found: Id'{selectedSection.Id}'")
+                };
+
+                model.Courses.Add(course.Id,thisCourse);
+
+                var baseSectionGroups = course.Sections.GroupBy(s => s.Meetings.First().Code[0]);
```

Comment:
Slight overuse of var where datatype declarations would be useful for cleanliness.